### DISCUSSION TITLE
插件的应用范围

### DISCUSSION THREAD

**Discussion by huangtugeone** [jd-opensource/drip-table#21] [General]
现在有antdesign这种插件好像已经可以实现表格的大部分功能，那啥时候需要用到这个，而且对于嵌套类型的json数据不知道支持程度如何，比如二级表头三级表头类似的（如果已经有实现记得@）

**Comment by helloqian12138** [comment]
是这样的，首先，我先回答一下drip-table和antd的Table组件的区别。
drip-table和antd的table组件是完全不一样的，主要区别有以下几点：
1、解决的痛点不一样。antd-table旨在简化前端Table的开发，封装了HTML原生Table；而drip-table是为了解决实际业务中Table的逻辑复杂难以维护的痛点，让用户通过低代码开发Table。
2、使用角色不一样。antd-table主要面向的还是前端开发人员；drip-table面向的使用者是产品人员和开发人员。
3、使用方式不一样。antd-table还是以组件形式进行引入，需要开发；但是drip-table可以帮助产品人员利用generator拖拖拽拽就可以生成表格，或者由开发人员进行LowCode化的组件开发。
4、接口不一样。antd-table提供的是组件化的接口；而drip-table提供的是一套渲染组件以及一套可视化生成器。

然后就是关于drip-table的使用场景：drip-table实现的是中后台LowCode化的CMS表格生成。
对于很多企业中后台平台或者应用，CMS的功能复杂，业务繁多。越到后面代码量越多，越来越难以维护。比如一个列表页包含几十上百种业务数据类型展示在一个表格内，这种情况即使利用antd的table组件实现，一列的渲染逻辑可能就成百上千行，更别提业务复杂后怎么去维护这里面的代码。而且对于后来的开发人员而言还得去读懂前人的代码逻辑，痛苦难当还影响了开发效率。
但是使用drip-table就可以对cms的表格渲染内容进行管理，一方面，产品人员可以通过可视化生成器自己去搭建表格；另一方面，业务开发人员只需要积累项目的业务组件即可。而且由于是LowCode开发的组件，所以每次开发也只需要关注自身需要实现的逻辑即可，不用去维护很复杂的代码逻辑，更不用去看别人的代码。

**Comment by helloqian12138** [comment]
至于你说的嵌套类型的json，麻烦详细说明一下，是指在单元格中嵌套新的表格内容吗？

**Comment by helloqian12138** [comment]
对于二级三级表头，感谢提出的宝贵意见，你可以提一个issue，然后我们会安排进日程。
